rest-api/data_gen/fake_participant_generator.py

```diff
@@ -295,38 +343,150 @@ def _make_measurement_resource(self, measurement, qualifier_set):
         } for qualifier in qualifiers]
     return resource
 
-  def _make_measurement_entry(self, measurement, time_str, participant_id, qualifier_set):
-    resource = self._make_measurement_resource(measurement, qualifier_set)
+  def _get_measurement_values(self, measurement_resources):
+    return [resource['valueQuantity']['value'] for resource in measurement_resources]
+
+  def _get_related(self, measurement_resources):
+    related_list = []
+    related_urls = set()
+    for measurement_resource in measurement_resources:
+      resource_related = measurement_resource.get('related')
+      if resource_related:
+        for related in resource_related:
+          # Make sure we don't repeat references to the same qualifier.
+          url = related['target']['reference']
+          if url not in related_urls:
+            related_list.append(related)
+            related_urls.add(url)
+    return related_list
+
+  def _mean(self, values):
+    return round(sum(values) / len(values), 1)
+
+  def _find_closest_two_mean(self, measurement_resources):
```

Comment:
Phew! Not proposing this because it ain't readable, but just found myself compelled to carry out the exercise. (I promise it shines when the # of repetitions is unbounded ;-))

```py
import itertools
def diff(xy): return abs(xy[0]-xy[1])

sorted_pairs = sorted(itertools.combinations(measurement_values, 2), key=diff)
grouped_pairs = itertools.groupby(sorted_pairs, diff)
best_group = next(grouped_pairs)[1]
closest_values = set(itertools.chain(*best_group))
self._mean(closest_values)
```